lib/policies/load-balancing.js

```diff
@@ -319,39 +336,32 @@ TokenAwareIterator.prototype.iterate = function (callback) {
 };
 
 TokenAwareIterator.prototype.computeNext = function () {
-  //return local replicas
-  var host = this.nextLocalReplica();
-  if (host) {
-    return {value: host, done: !host};
+  var host;
+  if (this.localReplicaIndex < this.localReplicas.length) {
+    host = this.localReplicas[(this.startIndex + (this.localReplicaIndex++)) % this.localReplicas.length];
+    return { value: host, done: false };
   }
-  //return child hosts
-  var item = this.childIterator.next();
-  if (!item.done) {
+  this.startIndex = -1;
+  // Return hosts from child policy
+  var item;
+  while ((item = this.childIterator.next()) && !item.done) {
     if (this.replicaMap[item.value.address]) {
-      return this.computeNext();
+      // Avoid yielding replicas from the child load balancing policy query plan
```

Comment:
This isn't entirely related to this PR, but was wondering what you thought about this.  It looks like this ordering logic is the same as the previous implementation (though easier to follow).  

However, couldn't the child policy return non-replica remote hosts causing this to prioritize non-replica remote hosts before replica remote hosts?

The ordering seems like this:

1. Yield local replica hosts.
2. Yield non-replica hosts.
3. Yield remote replica hosts.

The java and python drivers behaves slightly differently:

1. Yield local replica hosts.
2. Delegate to child policy and filter out hosts that are not local replicas.

I think the optimal solution would be something like:

1. Yield local replica hosts.
2. Yield non-replica local hosts.
3. Yield remote replica hosts.
4. Yield non-replica remote hosts.

Although that seems kind of tricky because you would need to keep a separate structure to track non-replica remote hosts while iterating in step 2.  Also I think the benefits of prioritizing remote replica hosts are probably marginal once you only have remote replicas to choose, so maybe the java/python solution is probably the most simple/effective.